assets/js/base/context/cart-checkout/payment-methods/use-payment-method-registration.js

```diff
@@ -86,8 +94,13 @@ const usePaymentMethodRegistration = (
 				[ paymentMethod.name ]: paymentMethod,
 			};
 		};
-		for ( const paymentMethodName in registeredPaymentMethods ) {
+
+		for ( let i = 0; i < paymentMethodsOrder.length; i++ ) {
```

Comment:
Nit: you can use `forEach` if you want:
```JS
await paymentMethodsOrder.forEach( async ( paymentMethodName ) => {
  ...
} );
```

Below, you would need to change the `continue` with `return`. Not blocking, just a suggestion in case you think it would make code cleaner. :slightly_smiling_face:

Reply:
This breaks! (Just tried it.) The [`foreach` doesn't wait for the async callback](https://advancedweb.hu/how-to-use-async-functions-with-array-foreach-in-javascript/), so nothing gets registered. 

I could switch it to `map` and `Promise.all`, not sure if that's clearer than the traditional indexed for loop. Thoughts @nerrad  @Aljullu ? Happy to change to `map` if that's more appropriate or more clear.

Reply:
Interesting! I'm fine with leaving the code as is. The for loop is clearer to me than using Array.map with Promise.all would be here.